File: lab3/F.py
# F - Make Them Odd

def choices(x):
    ch = []
    while x % 2 == 0:
        ch.append(x)
        x /= 2
    return ch

for i in range(int(input())):
    input()
    ns = map(int, input().split())
    ch = set()
    for j in ns:
            ch = ch.union(set(choices(j)))
    print(len(ch))

File: lab3/F.cpp
#include <iostream>
#include <set>
// F - Make Them Odd
using namespace std;

int main(){
    int t;
    cin >> t;
    for(int i = 0; i < t; i++){
        int n;
        cin >> n;
        set<int> s;
        for(int j = 0; j < n; j++){
            int k;
            cin >> k;
            while(k % 2 == 0){
                s.insert(k);
                k /= 2;
            }
        }
        cout << s.size() << endl;
    }
    return 0;
}
